### PR TITLE
Remove unstable feature and code and modernize

### PR DESCRIPTION
**Pull request by eldruin**
The tests can now be run on Rust >= 1.34.0 (due to Infallible).
Closes #28

### CODE REVIEW

**Comment by eldruin** [comment]
Can I get a review here @rust-embedded/hal ?
BTW I discovered that the commit of the 0.1.2 release was not merged into master. I have done that here as well.

**Comment by therealprof** [comment]
@eldruin As mentioned here: https://github.com/rust-embedded/nb/pull/28#issuecomment-651970273 we said we would wait a few days for objections before executing.

File: src/lib.rs [hunk 1]
```diff
@@ -57,29 +57,24 @@
 //! }
 //! ```
 //!
-//! You can use the *never type* (`!`) to signal that some API has no fatal
+//! You can use `Infallible` to signal that some API has no fatal
```
Comment (therealprof) [inline src/lib.rs:60]:
```suggestion
//! You can use `Infallible` type to signal that some API has no fatal
```

**Comment by eldruin** [comment]
Great. Should be solved now.

**Comment by therealprof** [comment]
Would you like to keep the 11 commits or condense this a bit more? (No request, just a question)

I think we'll wait until tomorrow and then I'll pull the trigger.

**Comment by eldruin** [comment]
Hmm I could squash them but then we would lose the info about the merged 0.1.2 release commit, since I discovered that too late.
I could squash around that in a new branch with cherry picking and so on but honestly it feels like busy work.

**Comment by therealprof** [comment]
@eldruin Hah, you could use that opportunity to learn about `git rebase -i`. ;)

**Comment by eldruin** [comment]
Ok I've squashed them now.

File: Cargo.toml [hunk 2]
```diff
@@ -9,10 +9,5 @@ repository = "https://github.com/rust-embedded/nb"
...
 documentation = "https://docs.rs/nb"
 readme = "README.md"
-version = "0.1.1"
-
-[features]
-unstable = []
-
-[dev-dependencies]
-futures = "0.1.17"
+version = "0.1.2"
```
Comment (therealprof) [inline Cargo.toml:12]: Wrong version. ;)

Comment (eldruin) [reply Cargo.toml]: Should be fixed now.

**Pull request merged**